RESTResource/RESTResource.js

```diff
@@ -661,7 +661,23 @@ export default class RESTResource {
                   httpStatus: response.status,
                   other: records ? _.omit(json, records) : {},
                 };
-                if (meta.other) meta.other.totalRecords = extractTotal(json);
+
+                if (meta.other) {
+                  const totalRecords = extractTotal(json);
+                  // Please see https://issues.folio.org/browse/STSMACOM-259
+                  // for more details.
+
+                  // The totalRecords returned from the backend are sometimes
+                  // incorrect and instead of returning the actual number they
+                  // return 999999 which causes an extra fetch which in turn
+                  // returns an empty result. This is a workaround for this issue.
+                  if (totalRecords === 0) {
+                    return;
+                  }
```

Comment:
The problem with just returning here is above at https://github.com/folio-org/stripes-connect/blob/8a239763f21089fa1d0e610d0c920d77092bb5dc/RESTResource/RESTResource.js#L313-L324
```
action.meta.url === val.url
```
will always return false 
because for a record set with ~100 rows, in the penultimate query we'll see values like 
```
...limit=30&offset=90 === ...limit=30&offset=120
```
That will trigger another query with `limit=30&offset=120`, but then we don't dispatch a `success` since this query will have the incorrect `totalRecords=0` attribute.

Reply:
This is a good catch @zburke! Thank you double checking this. I think we should probably close this PR.